test/benchmark/clock_gettime.rb

```diff
@@ -0,0 +1,27 @@
+# frozen_string_literal: true
+
+require 'benchmark/ips'
+
+Benchmark.ips do |bench|
+  bench.report("Process.clock_gettime in milliseconds (int)") do
+    Process.clock_gettime(Process::CLOCK_MONOTONIC, :millisecond)
+  end
+
+  bench.report("Process.clock_gettime in milliseconds (float)") do
+    Process.clock_gettime(Process::CLOCK_MONOTONIC, :float_millisecond)
+  end
+
+  bench.report("Process.clock_gettime in seconds (float), multiplied by 1000") do
+    1000 * Process.clock_gettime(Process::CLOCK_MONOTONIC)
+  end
+
+  bench.report("Process.clock_gettime in seconds (float), multiplied by 1000.0") do
+    1000.0 * Process.clock_gettime(Process::CLOCK_MONOTONIC)
```

Comment:
The underlying resolution is a struct that holds seconds & nanoseconds as separate members, stored as `time_t` & `long int`: https://linux.die.net/man/3/clock_gettime. Because of this underlying resolution, I am not too worried about additional* floating point error. (* = We have to convert this struct into a floating point number anyway, so it's already a lossy operation)

I think this also explains why `float_seconds` is faster than `float_microseconds`, because it's easier to convert this struct to a Float where seconds are the integer part. #dunnolol though

This should be the same for any CPU type. Because this is part of the POSIX standard, that characteristic should be true for all UNIX-like OSes. For Windows it may not be the case but I am happy to ignore that.